src/test/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumerTest.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.companieshouse.itemhandler.kafka;
+
+import org.hamcrest.Matchers;
+import org.junit.Assert;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.kafka.consumer.resilience.CHKafkaResilientConsumerGroup;
+import uk.gov.companieshouse.kafka.exceptions.SerializationException;
+
+import java.util.concurrent.ExecutionException;
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class OrdersKafkaConsumerTest {
+    private static final String ORDER_RECEIVED_URI = "/order/ORDER-12345";
+
+    @Spy
+    @InjectMocks
+    private OrdersKafkaConsumer ordersKafkaConsumer;
+    @Mock
+    private CHKafkaResilientConsumerGroup chKafkaConsumerGroupMain;
+    @Mock
+    private CHKafkaResilientConsumerGroup chKafkaConsumerGroupRetry;
+
+    @Test
+    public void mainListenerExceptionIsCorrectlyHandled() throws InterruptedException, ExecutionException, SerializationException {
+        // Given & When
+        doThrow(new OrderProcessingException()).when(ordersKafkaConsumer).processOrderReceived(ORDER_RECEIVED_URI);
+        Exception exception = Assertions.assertThrows(RuntimeException.class, () -> {
```

Comment:
* Is the type of the exception deliberately looser here than in the other two tests?
* If not, these three methods seem to share very similar logic, perhaps some scope for refectoring?